### PR TITLE
Update dependencies to support Apple M1, fix Windows CI LLVM paths

### PR DESCRIPTION
**Pull request by fanatid**
## Motivation

Ublock development on Apple M1.

## Solution

Update dependencies.

## Related Issues

- https://github.com/ZcashFoundation/zcash_script/pull/20
- https://github.com/rust-rocksdb/rust-rocksdb/pull/506#issuecomment-814675349

### CODE REVIEW

**Comment by fanatid** [comment]
Same windows clang issue as in zcash_script https://github.com/ZcashFoundation/zcash_script/pull/20 🤦 Will do proper fix there first.

File: Cargo.toml
Comment (teor2345) [inline Cargo.toml]: Let's update this git dependency to the Foundation's repository before we merge.

(We try to avoid depending on too many git repositories, because the build will break if someone accidentally deletes one of the commits we're depending on.)

Comment (fanatid) [reply Cargo.toml]: I think I needed originally to create a draft PR. Would be possible to release a new version of `zcash_script` (once https://github.com/ZcashFoundation/zcash_script/pull/20 will be merged)?

Comment (teor2345) [reply Cargo.toml]: We can reference a commit in the Foundation's git repository as soon as we merge your zcash_script PR.

We can also do a release, but we shouldn't block this PR on doing a release.

**Comment by teor2345** [comment]
It looks like it might be a while before GitHub has M1 CI runners:
https://github.com/actions/virtual-environments/issues/2187

But we can test manually until then.

**Comment by fanatid** [comment]
Yeah, M1 still not publicity available in GitHub CI 😞
I think it's okay for some time do not to test on M1 in CI. At least these changes allow to use M1 for development.

File: Cargo.toml
Comment (dconnolly) [reply Cargo.toml]: > I think I needed originally to create a draft PR. Would be possible to release a new version of `zcash_script` (once [ZcashFoundation/zcash_script#20](https://github.com/ZcashFoundation/zcash_script/pull/20) will be merged)?

zcash_script@0.1.6-alpha.0 has been released on crates.io 📦

File: Cargo.toml
Comment (dconnolly) [inline Cargo.toml]:
```suggestion
```

File: zebra-script/Cargo.toml
Comment (dconnolly) [inline zebra-script/Cargo.toml:11]: 👍

File: zebra-state/Cargo.toml
Comment (dconnolly) [inline zebra-state/Cargo.toml:26]: 👍

**Comment by fanatid** [comment]
Updated to latest crates 🎉

**Review by teor2345 — CHANGES_REQUESTED** [review]
Let's fix the hard-coded block name, and then we're good to merge.

You can also move the lexical-core change to #2026 if you want, but I don't want to block on that.

**Comment by teor2345** [comment]
> Let's fix the hard-coded block name, and then we're good to merge.
> 
> You can also move the lexical-core change to #2026 if you want, but I don't want to block on that.

Oops sorry this review was meant for #2018. I'll just go fix that all up now.

**Review by teor2345 — APPROVED** [review]
Looks like we're good to merge here.

**Comment by teor2345** [comment]
Thanks for all your hard work on this PR, and the zcash_script PR!

**Comment by teor2345** [comment]
I updated the title to describe what the merged PR actually does, so our changelog is accurate.

There's no need to match the dependabot title format for future PRs. Having different PR titles from dependabot actually helps us identify manual changes in the changelog.

**Pull request merged**
